connectors/ssh/src/main/java/org/crsh/ssh/term/SSHFactories.java

```diff
@@ -67,23 +67,27 @@ public class SSHFactories {
                             BuiltinCompressions.none
                     ));
 
+    @SuppressWarnings({ "unchecked", "rawtypes" }) // safe due to the hierarchy
```

Comment:
The suppressions and type-castings in this and the subsequent methods are the result of peeking into the changes made to Apache SSHD - this is the sort of thing that was put in their code in the same PR where the the return-type of _NamedFactory.setUpBuiltinFactories()_ was changed. Look at [the changes in ClientBuilder.java](https://github.com/apache/mina-sshd/commit/f82e0db2ca23f806d8e31ced9156132f76017483#diff-775f133e266d3edc40c0cdb2ec199a77acfca08011715bc29f51722f7663f5db), line 196 onwards.